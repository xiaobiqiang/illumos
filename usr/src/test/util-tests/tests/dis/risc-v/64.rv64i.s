/*
 * This file and its contents are supplied under the terms of the
 * Common Development and Distribution License ("CDDL"), version 1.0.
 * You may only use this file in accordance with the terms of version
 * 1.0 of the CDDL.
 *
 * A full copy of the text of the CDDL should have accompanied this
 * source.  A copy of the CDDL is also available via the Internet at
 * http://www.illumos.org/license/CDDL.
 */

/*
 * Copyright 2018, Joyent, Inc.
 */

/*
 * Test our disassembly of the RV64I instructions. Instructions are ordered per the
 * ISA manual.
 */

.text
.align 16
.globl libdis_test
.type libdis_test, @function
libdis_test:
	lwu	s7, 0x7ff(s1)
	lwu	s6, (s1)
	lwu	s5, -0x800(s1)
	ld	s4, 0x7ff(s1)
	ld	s3, (s1)
	ld	s2, -0x800(s1)
	sd	t0, 0x7ff(t1)
	sd	t1, (t2)
	sd	t2, -0x800(t3)
	slli	s0, s1, 2
	slli	s0, s1, 63
	srli	s0, s1, 2
	srli	s0, s1, 63
	srai	s0, s1, 2
	srai	s0, s1, 63
	addiw	ra, t0, 0x4
	addiw	ra, t0, -0x4
	slliw	t4, t5, 0x12
	srliw	t4, t5, 0x13
	sraiw	t4, t5, 0x14
	addw	s0, s1, s2
	subw	s1, s2, s3
	sllw	s3, s4, s5
	srlw	s3, s4, s5
	sraw	s3, s4, s5

.size libdis_test, [.-libdis_test]
