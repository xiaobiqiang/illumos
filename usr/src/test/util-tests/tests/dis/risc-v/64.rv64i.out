    libdis_test:      83 eb f4 7f        lwu s7,0x7ff(s1)
    libdis_test+0x4:  03 eb 04 00        lwu s6,0x0(s1)
    libdis_test+0x8:  83 ea 04 80        lwu s5,-0x800(s1)
    libdis_test+0xc:  03 ba f4 7f        ld s4,0x7ff(s1)
    libdis_test+0x10: 83 b9 04 00        ld s3,0x0(s1)
    libdis_test+0x14: 03 b9 04 80        ld s2,-0x800(s1)
    libdis_test+0x18: a3 3f 53 7e        sd t0,0x7ff(t1)
    libdis_test+0x1c: 23 b0 63 00        sd t1,0x0(t2)
    libdis_test+0x20: 23 30 7e 80        sd t2,-0x800(t3)
    libdis_test+0x24: 13 94 24 00        slli s0,s1,0x2
    libdis_test+0x28: 13 94 f4 03        slli s0,s1,0x3f
    libdis_test+0x2c: 13 d4 24 00        srli s0,s1,0x2
    libdis_test+0x30: 13 d4 f4 03        srli s0,s1,0x3f
    libdis_test+0x34: 13 d4 24 40        srai s0,s1,0x2
    libdis_test+0x38: 13 d4 f4 43        srai s0,s1,0x3f
    libdis_test+0x3c: 9b 80 42 00        addiw ra,t0,0x4
    libdis_test+0x40: 9b 80 c2 ff        addiw ra,t0,-0x4
    libdis_test+0x44: 9b 1e 2f 01        slliw t4,t5,0x12
    libdis_test+0x48: 9b 5e 3f 01        srliw t4,t5,0x13
    libdis_test+0x4c: 9b 5e 4f 41        sraiw t4,t5,0x14
    libdis_test+0x50: 3b 84 24 01        addw s0,s1,s2
    libdis_test+0x54: bb 04 39 41        subw s1,s2,s3
    libdis_test+0x58: bb 19 5a 01        sllw s3,s4,s5
    libdis_test+0x5c: bb 59 5a 01        srlw s3,s4,s5
    libdis_test+0x60: bb 59 5a 41        sraw s3,s4,s5
